src/category_theory/limits/shapes/constructions/finite_products_of_binary_products.lean

```diff
@@ -0,0 +1,210 @@
+/-
+-- Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Bhavik Mehta
+-/
+import category_theory.limits.shapes.finite_products
+import category_theory.limits.shapes.binary_products
+import category_theory.limits.preserves.shapes.products
+import category_theory.limits.preserves.shapes.binary_products
+import category_theory.limits.shapes.pullbacks
+import category_theory.pempty
+import data.equiv.fin
+
+/-!
+# Constructing finite products from binary products and terminal.
+
+If a category has binary products and a terminal object then it has finite products.
+If a functor preserves binary products and the terminal object then it preserves finite products.
+
+# TODO
+
+Provide the dual results.
+Show the analogous results for functors which reflect or create (co)limits.
+-/
+
+universes v u u'
+
+noncomputable theory
+open category_theory category_theory.category category_theory.limits
+namespace category_theory
+
+variables {J : Type v} [small_category J]
+variables {C : Type u} [category.{v} C]
+variables {D : Type u'} [category.{v} D]
+
+/--
+Given `n+1` objects of `C`, a fan for the last `n` with point `c₁.X` and a binary fan on `c₁.X` and
+`f 0`, we can build a fan for all `n+1`.
+
+In `build_limit` we show that if the two given fans are limits, then this fan is also a limit.
+-/
+@[simps {rhs_md := semireducible}]
+def build_prod {n : ℕ} {f : ulift (fin (n+1)) → C}
```

Comment:
Should this maybe be called `extend_fan` or `fan_succ` or `fan_of_fan_of_binary_fan`?

Reply:
Sounds good, amended.